### PR TITLE
[5.2] Sending to a Specified Address (Global "to")

### PR DESCRIPTION
**Pull request by clemblanco**
Sometimes you want to avoid email real customers. The two solutions given by Laravel as per as the official documentation are:
- Using "pretend" to log the emails into the log file.
- Using something like http://mailtrap.io.

Like with [Symfony2](http://symfony.com/doc/current/cookbook/email/dev_environment.html#sending-to-a-specified-address), here we also give the ability to the developer to specify one email address which will receive every single email from his application. Could be very useful on a staging environment where you don't want to email real customers and where the actual tester wouldn't have access to the log files or wouldn't be able to use mailtrap.

### CODE REVIEW

**Comment by taylorotwell** [comment]
Wouldn't using a ->to() method call in your Mail::send override or at least append to this?

File: src/Illuminate/Mail/Mailer.php [hunk 1]
```diff
@@ -112,6 +112,18 @@ public function alwaysFrom($address, $name = null)
 	}
 
 	/**
+	 * Set the global to address and name.
+	 *
+	 * @param  string  $address
+	 * @param  string  $name
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Mailer.php]: string|null

**Comment by Claymm** [comment]
I don't want to append, I would like to override.
I don't want to change my codebase for that, I would like this to be set at the environment level.
Just realized that this is not complete. Will push another PR. Please close this one.

**Comment by Claymm** [comment]
Ok this has been tested and it works as I expected. If I setup the 'to' array in any mail.php config file for a specific environment, all emails are sent to this address.

File: src/Illuminate/Mail/Mailer.php (hunk 1 above)
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Mailer.php]: The needs changing to `string|null`. `string` is wrong.

Comment (Claymm) [inline src/Illuminate/Mail/Mailer.php]: Well, was already wrong initially on the alwaysFrom() function then as I got it from there. Will do that on both.

Comment (GrahamCampbell) [inline src/Illuminate/Mail/Mailer.php]: Both are wrong then, yeh.

Comment (lukasgeiter) [inline src/Illuminate/Mail/Mailer.php]: @GrahamCampbell Parameters that default to null are documented like that in many places in the framework. Just one example: [Query Builder](https://github.com/laravel/framework/blob/5.0/src/Illuminate/Database/Query/Builder.php#L312-L323)

**Comment by GrahamCampbell** [comment]
Please fix the merge conflicts.

**Comment by Claymm** [comment]
Yep my bad I forked this ages ago. Was in 4.2. I need to get my fork up to date first. Will do it later...

**Comment by Claymm** [comment]
Should be alright now. Let me know if there is any issue.

File: src/Illuminate/Mail/Mailer.php [hunk 2]
```diff
@@ -103,8 +103,8 @@ public function __construct(Factory $views, Swift_Mailer $swift, Dispatcher $eve
 	/**
 	 * Set the global from address and name.
 	 *
-	 * @param  string  $address
-	 * @param  string  $name
+	 * @param  string       $address
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Mailer.php]: Don't align this please - just use 2 spaces.

File: src/Illuminate/Mail/Mailer.php [hunk 3]
```diff
@@ -113,6 +113,18 @@ public function alwaysFrom($address, $name = null)
 	}
 
 	/**
+	 * Set the global to address and name.
+	 *
+	 * @param  string       $address
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Mailer.php]: Same here.

File: src/Illuminate/Mail/Message.php [hunk 4]
```diff
@@ -68,11 +68,16 @@ public function returnPath($address)
 	 * Add a recipient to the message.
 	 *
 	 * @param  string|array  $address
-	 * @param  string  $name
+	 * @param  string|null   $name
+	 * @param  boolean       $override   Will force ignoring the previous recipients
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Message.php]: bool not boolean, and don't align it please.

Comment (GrahamCampbell) [inline src/Illuminate/Mail/Message.php]: Also, could you move this argument description to be the "Long description" please.

File: src/Illuminate/Mail/Message.php [hunk 5]
```diff
@@ -68,11 +68,16 @@ public function returnPath($address)
...
 	 * @return $this
 	 */
-	public function to($address, $name = null)
+	public function to($address, $name = null, $override = false)
 	{
+		if ($override) {
```
Comment (GrahamCampbell) [inline src/Illuminate/Mail/Message.php]: brace on a newline plz

File: src/Illuminate/Mail/Message.php (hunk 4 above)
Comment (Claymm) [inline src/Illuminate/Mail/Message.php]: What do you mean?

**Comment by GrahamCampbell** [comment]
Thank you @Claymm. We've applied this to 5.1.

**Pull request closed without merging**